elasticgraph-datastore_core/spec/integration/elastic_graph/datastore_core/index_definition/implementation_shared_examples.rb

```diff
@@ -17,59 +17,46 @@ module IndexDefinition
           it "returns `false` on an index that has no `sourced_from` fields" do
             index = define_index
 
-            expect {
-              expect(index.searches_could_hit_incomplete_docs?).to be false
-            }.to change { datastore_requests("main").count }.by(1)
-
-            # Demonstrate that we cache the value by showing the datastore request count doesn't change
-            # when we call `searches_could_hit_incomplete_docs?` with the same client again.
-            expect {
-              expect(index.searches_could_hit_incomplete_docs?).to be false
-            }.not_to change { datastore_requests("main").count }
+            expect(index.searches_could_hit_incomplete_docs?).to be false
           end
 
-          it "returns `true` on an index that has `sourced_from` fields, without hitting the datastore (since it is not needed!)" do
-            index = define_index do |t|
+          it "returns `true` on an index that has `sourced_from` fields" do
+            index = define_index(has_sourced_from: true) do |t|
               t.field "owner_name", "String" do |f|
                 f.sourced_from "owner", "name"
               end
             end
 
-            expect {
-              expect(index.searches_could_hit_incomplete_docs?).to be true
-            }.not_to change { datastore_requests("main").count }
-
-            # Demonstrate that we cache the value by showing the datastore request count doesn't change
-            # when we call `searches_could_hit_incomplete_docs?` with the same client again.
-            expect {
-              expect(index.searches_could_hit_incomplete_docs?).to be true
-            }.not_to change { datastore_requests("main").count }
+            expect(index.searches_could_hit_incomplete_docs?).to be true
           end
 
-          it "returns `true` on an index that no longer has `sourced_from` fields but used to" do
-            define_index do |t|
+          it "returns `true` on an index that no longer has `sourced_from` fields but the `has_had_multiple_sources!` flag remains set" do
```

Comment:
At this point I don't think `sourced_from` really has any bearing on the behavior of `searches_could_hit_incomplete_docs?`.  It's based purely on whether `has_had_multiple_sources!` is currently called on the index.

I don't see a way for this test to fail without causing the `it "uses the `has_had_multiple_sources` flag from runtime metadata" do` test to fail...so it's not clear to me this test needs to exist.